pubnub/event_engine/manage_effects.py

```diff
@@ -0,0 +1,138 @@
+import asyncio
+
+from queue import SimpleQueue
+from typing import Union
+from pubnub.endpoints.pubsub.subscribe import Subscribe
+from pubnub.pubnub import PubNub
+from pubnub.event_engine.models import effects, events
+from pubnub.models.consumer.common import PNStatus
+from pubnub.workers import SubscribeMessageWorker
+
+
+class ManagedEffect:
+    pubnub: PubNub = None
+    event_engine = None
+    effect: Union[effects.PNManageableEffect, effects.PNCancelEffect]
+
+    def set_pn(self, pubnub: PubNub):
+        self.pubnub = pubnub
+
+    def __init__(self, pubnub_instance, event_engine_instance,
+                 effect: Union[effects.PNManageableEffect, effects.PNCancelEffect]) -> None:
+        self.effect = effect
+        self.event_engine = event_engine_instance
+        self.pubnub = pubnub_instance
+
+    def run(self):
+        pass
+
+    def run_async(self):
```

Comment:
Is this used anywhere?

Reply:
`run` method is a placeholder for future synchronous implementation